CRITICAL - 2024-02-01 02:00:29 --> ErrorException: Undefined index: id_satuan
in APPPATH\Views\satuan\data_satuan.php on line 54.
 1 APPPATH\Views\satuan\data_satuan.php(54): CodeIgniter\Debug\Exceptions->errorHandler(8, 'Undefined index: id_satuan', 'C:\\xampp\\htdocs\\ci4\\app\\Views\\satuan\\data_satuan.php', 54, [...])
 2 SYSTEMPATH\View\View.php(228): include('C:\\xampp\\htdocs\\ci4\\app\\Views\\satuan\\data_satuan.php')
 3 SYSTEMPATH\View\View.php(231): CodeIgniter\View\View->CodeIgniter\View\{closure}()
 4 SYSTEMPATH\Common.php(1178): CodeIgniter\View\View->render('satuan/data_satuan', [], true)
 5 APPPATH\Controllers\Satuan.php(14): view('satuan/data_satuan', [...])
 6 SYSTEMPATH\CodeIgniter.php(941): App\Controllers\Satuan->index()
 7 SYSTEMPATH\CodeIgniter.php(502): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Satuan))
 8 SYSTEMPATH\CodeIgniter.php(361): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 FCPATH\index.php(79): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Commands\Server\rewrite.php(47): require_once('C:\\xampp\\htdocs\\ci4\\public\\index.php')
ERROR - 2024-02-01 04:25:04 --> Error connecting to the database: mysqli_sql_exception: No connection could be made because the target machine actively refused it.
 in C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\Database\MySQLi\Connection.php:185
Stack trace:
#0 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\Database\MySQLi\Connection.php(185): mysqli->real_connect('localhost', 'root', '', 'ci4', 3306, '', 0)
#1 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\Database\BaseConnection.php(392): CodeIgniter\Database\MySQLi\Connection->connect(false)
#2 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\Database\BaseConnection.php(575): CodeIgniter\Database\BaseConnection->initialize()
#3 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\Database\BaseBuilder.php(1613): CodeIgniter\Database\BaseConnection->query('SELECT *\nFROM `...', Array, false)
#4 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\Model.php(242): CodeIgniter\Database\BaseBuilder->get()
#5 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\BaseModel.php(638): CodeIgniter\Model->doFindAll(0, 0)
#6 C:\xampp\htdocs\ci4\app\Controllers\Kategori.php(12): CodeIgniter\BaseModel->findAll()
#7 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\CodeIgniter.php(941): App\Controllers\Kategori->index()
#8 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\CodeIgniter.php(502): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Kategori))
#9 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\CodeIgniter.php(361): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#10 C:\xampp\htdocs\ci4\public\index.php(79): CodeIgniter\CodeIgniter->run()
#11 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(47): require_once('C:\\xampp\\htdocs...')
#12 {main}

Next CodeIgniter\Database\Exceptions\DatabaseException: No connection could be made because the target machine actively refused it.
 in C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\Database\MySQLi\Connection.php:229
Stack trace:
#0 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\Database\BaseConnection.php(392): CodeIgniter\Database\MySQLi\Connection->connect(false)
#1 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\Database\BaseConnection.php(575): CodeIgniter\Database\BaseConnection->initialize()
#2 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\Database\BaseBuilder.php(1613): CodeIgniter\Database\BaseConnection->query('SELECT *\nFROM `...', Array, false)
#3 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\Model.php(242): CodeIgniter\Database\BaseBuilder->get()
#4 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\BaseModel.php(638): CodeIgniter\Model->doFindAll(0, 0)
#5 C:\xampp\htdocs\ci4\app\Controllers\Kategori.php(12): CodeIgniter\BaseModel->findAll()
#6 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\CodeIgniter.php(941): App\Controllers\Kategori->index()
#7 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\CodeIgniter.php(502): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Kategori))
#8 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\CodeIgniter.php(361): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#9 C:\xampp\htdocs\ci4\public\index.php(79): CodeIgniter\CodeIgniter->run()
#10 C:\xampp\htdocs\ci4\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(47): require_once('C:\\xampp\\htdocs...')
#11 {main}
CRITICAL - 2024-02-01 04:25:04 --> CodeIgniter\Database\Exceptions\DatabaseException: Unable to connect to the database.
Main connection [MySQLi]: No connection could be made because the target machine actively refused it.

in SYSTEMPATH\Database\BaseConnection.php on line 428.
 1 SYSTEMPATH\Database\BaseConnection.php(575): CodeIgniter\Database\BaseConnection->initialize()
 2 SYSTEMPATH\Database\BaseBuilder.php(1613): CodeIgniter\Database\BaseConnection->query('SELECT *
FROM `kategori`', [], false)
 3 SYSTEMPATH\Model.php(242): CodeIgniter\Database\BaseBuilder->get()
 4 SYSTEMPATH\BaseModel.php(638): CodeIgniter\Model->doFindAll(0, 0)
 5 APPPATH\Controllers\Kategori.php(12): CodeIgniter\BaseModel->findAll()
 6 SYSTEMPATH\CodeIgniter.php(941): App\Controllers\Kategori->index()
 7 SYSTEMPATH\CodeIgniter.php(502): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Kategori))
 8 SYSTEMPATH\CodeIgniter.php(361): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 FCPATH\index.php(79): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Commands\Server\rewrite.php(47): require_once('C:\\xampp\\htdocs\\ci4\\public\\index.php')
CRITICAL - 2024-02-01 15:00:24 --> ErrorException: Undefined index: katnama
in APPPATH\Views\produk\data_produk.php on line 60.
 1 APPPATH\Views\produk\data_produk.php(60): CodeIgniter\Debug\Exceptions->errorHandler(8, 'Undefined index: katnama', 'C:\\xampp\\htdocs\\ci4\\app\\Views\\produk\\data_produk.php', 60, [...])
 2 SYSTEMPATH\View\View.php(228): include('C:\\xampp\\htdocs\\ci4\\app\\Views\\produk\\data_produk.php')
 3 SYSTEMPATH\View\View.php(231): CodeIgniter\View\View->CodeIgniter\View\{closure}()
 4 SYSTEMPATH\Common.php(1178): CodeIgniter\View\View->render('produk/data_produk', [], true)
 5 APPPATH\Controllers\Produk.php(13): view('produk/data_produk', [...])
 6 SYSTEMPATH\CodeIgniter.php(941): App\Controllers\Produk->index()
 7 SYSTEMPATH\CodeIgniter.php(502): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Produk))
 8 SYSTEMPATH\CodeIgniter.php(361): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 FCPATH\index.php(79): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Commands\Server\rewrite.php(47): require_once('C:\\xampp\\htdocs\\ci4\\public\\index.php')
CRITICAL - 2024-02-01 15:20:52 --> CodeIgniter\Database\Exceptions\DataException: There is no data to insert.
in SYSTEMPATH\BaseModel.php on line 788.
 1 SYSTEMPATH\BaseModel.php(788): CodeIgniter\Database\Exceptions\DataException::forEmptyDataset('insert')
 2 SYSTEMPATH\Model.php(744): CodeIgniter\BaseModel->insert([], true)
 3 APPPATH\Controllers\Produk.php(26): CodeIgniter\Model->insert([...])
 4 SYSTEMPATH\CodeIgniter.php(941): App\Controllers\Produk->simpanProduk()
 5 SYSTEMPATH\CodeIgniter.php(502): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Produk))
 6 SYSTEMPATH\CodeIgniter.php(361): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 7 FCPATH\index.php(79): CodeIgniter\CodeIgniter->run()
 8 SYSTEMPATH\Commands\Server\rewrite.php(47): require_once('C:\\xampp\\htdocs\\ci4\\public\\index.php')
